code/io/joy-sdl.cpp

```diff
@@ -800,6 +800,42 @@ namespace joystick
 
 		SDL_QuitSubSystem(SDL_INIT_JOYSTICK);
 	}
+
+	SCP_vector<JoystickInformation> getJoystickInformations() {
+		SCP_vector<JoystickInformation> joysticks;
+
+		if (SDL_InitSubSystem(SDL_INIT_JOYSTICK) < 0) {
+			return SCP_vector<JoystickInformation>();
+		}
+
+		auto num_joysticks = SDL_NumJoysticks();
+		for (auto i = 0; i < num_joysticks; ++i) {
+			auto joystick = SDL_JoystickOpen(i);
+
+			JoystickInformation info{};
+
+			auto name = SDL_JoystickName(joystick);
+			if (name != nullptr) {
+				info.name = name;
+			}
+			info.guid = getJoystickGUID(joystick);
+
+			info.num_axes = static_cast<uint32_t>(SDL_JoystickNumAxes(joystick));
+			info.num_balls = static_cast<uint32_t>(SDL_JoystickNumBalls(joystick));
+			info.num_buttons = static_cast<uint32_t>(SDL_JoystickNumButtons(joystick));
+			info.num_hats = static_cast<uint32_t>(SDL_JoystickNumHats(joystick));
+
+			info.is_haptic = static_cast<bool>(SDL_JoystickIsHaptic(joystick));
```

Comment:
This line is generating a warning in VS from the `static_cast<bool>`; you need to either use one of the other paradigms for saying "yes, I really want this to become a boolean value" like a double-not, or change `SDL_JoystickIsHaptic()` to return a `bool`.

Reply:
How would I change `SDL_JoystickIsHaptic()` to return a `bool`? It's a C API so it doesn't even have that type and it's also an external API so it's absolutely impossible to change that code.

I fixed it by comparing the return value with `SDL_TRUE` which should be the safest way of converting the result into a `bool`.